src/main/java/org/broadinstitute/hellbender/tools/exome/AllelicCountCollection.java

```diff
@@ -33,10 +34,14 @@ public AllelicCountCollection() {
      * @param inputFile     file to read from
      */
     public AllelicCountCollection(final File inputFile) {
+        Utils.nonNull(inputFile);
+        Utils.regularReadableUserFile(inputFile);
         try (final TableReader<AllelicCount> reader = TableUtils.reader(inputFile,
                 (columns, formatExceptionFactory) -> {
-                    if (!columns.matchesExactly(AllelicCountTableColumns.COLUMN_NAME_ARRAY))
-                        throw formatExceptionFactory.apply("Bad header");
+                    if (!columns.containsAll(AllelicCountTableColumns.COLUMN_NAME_ARRAY)) {
```

Comment:
This pattern appears in several places.  Why not extract a utility method that checks for all required columns and throws an error with the missing columns?  This method would have a signature like `checkRequiredColumns(final TableColumnCollection columns, final Function<String, RuntimeException> formatExceptionFactory, final Collection<String> requiredColumnNames)`.  Honestly if it were up to me I'd go a little further and replace arbitrarily-typed `formatExceptionFactory` in `TableReader` with some fixed exception, eg `MissingColumnsException`, but that's beyond the scope of this PR.

Places where this refactoring could simplify code:
* `CoverageDropoutResult.java` line 172
* `SegmentUtils.java` line 426
* `TargetTableReader.java` line 51
* `SampleCoverageStatsReader.java` line 46